### PR TITLE
Replaced deprecated xblock fragment with web_fragment

### PR DESCRIPTION
**Pull request by mraarif**
#### What are the relevant tickets?
https://openedx.atlassian.net/browse/BOM-1862

#### What's this PR do?
remove deprecated usage of `xblock.fragment`

### CODE REVIEW

**Comment by mraarif** [comment]
@jmbowman latest version of `xblock` only supports python 3, shouldn't we plan removing python 2 support here as well?

**Comment by mraarif** [comment]
@pdpinch can you get the review started on this?

**Comment by pdpinch** [comment]
@mraarif can you rebase?

**Comment by mraarif** [comment]
@pdpinch done

**Comment by pdpinch** [comment]
do you expect this to work in Juniper as well as the edx-platform master branch?

Reviews in master are getting held up due to a devstack bug.

**Comment by mraarif** [comment]
I can create a PR (to test) in `edx-platform` targeting this commit instead of fetching `edx-sga` from PyPi. @jmbowman this should be enough to test these changes in`edxplatform`?

**Comment by jmbowman** [comment]
I suspect there isn't much specific testing of edx-sga in edx-platform, so they probably want to do some manual testing.  Which devstack bug is causing the delay?

**Comment by pdpinch** [comment]
I don't know if there's a ticket for the devstack issue, but there's some discussion on Slack here: https://openedx.slack.com/archives/CDAG4KN2C/p1600356900015400

**Comment by pdpinch** [comment]
@mraarif can you help me understand the scope of this PR? The issue seems quite narrow, but it looks like you've added a lot of boilerplate code, and also cleaned up the linting.

**Comment by mraarif** [comment]
the change I made was replacing `Xblock.fragment` with `web-fragment` and its usage, I had to install `web_fragment` and I took the liberty to introduce `pip-tools` to manage dependencies and move dependencies out of `setup.py` that caused the creation of all `.txt` files, and after packages upgrade, new versions failed due to quality so I had to do a little cleanup.

**Comment by jmbowman** [comment]
@pdpinch Let us know if you think we should revert the dependency management changes.  It's for compliance with OEP-18 and we tend to do that instinctively now before doing other package upgrades because it prevents some common pitfalls, but this isn't an edX-owned repo so it's your call on whether you want it set up this way.

**Comment by asadiqbal08** [comment]
@mraarif the build is failing over here, can you please address the failure over this PR ?

**Comment by mraarif** [comment]
@asadiqbal08 coverage is failing due to the refactoring we did, we can revert this as @jmbowman said and go back to have only web-fragment changes that should fix coverage failure.

**Comment by pdpinch** [comment]
Thank you for offering @mraarif -- please revert to just the web-fragment changes so we can prioritize those. 

If you'd like to, please open a separate PR for the dependency management changes, so we can review them separately.

**Comment by mraarif** [comment]
@pdpinch @asadiqbal08 this should be ready for a re-review now, I still had to remove some unused-imports to get quality checks to pass

File: edx_sga/__init__.py [hunk 1]
```diff
@@ -2,5 +2,4 @@
 Module for StaffGradedAssignmentXBlock.
 """
 
-__version__ = '0.11.1'
-
+__version__ = '0.12.0'
```
Comment (asadiqbal08) [inline edx_sga/__init__.py:5]: @mraarif Doof will do the versioning for us so do not need it in the PR.

Comment (mraarif) [reply edx_sga/__init__.py]: reverted

**Comment by asadiqbal08** [comment]
@pdpinch I tested and seems me working fine, do u want to add another sight over that PR ?

File: edx_sga/__init__.py [hunk 2]
```diff
@@ -2,5 +2,4 @@
 Module for StaffGradedAssignmentXBlock.
 """
 
-__version__ = '0.11.1'
-
+__version__ = '0.10.0'
```
Comment (pdpinch) [inline edx_sga/__init__.py:5]: Can you remove this version number change? We prefer to handle this with our release bot.

Comment (mraarif) [reply edx_sga/__init__.py]: done

**Comment by pdpinch** [comment]
Thank you for your submission @mraarif and for your patience. This looks good to me. I'll merge it today. I think we'll do a release tomorrow after #278 is merged. 

Do you plan to open another PR for the dependency management tooling you removed from this PR?

**Comment by awais786** [comment]
@pdpinch @asadiqbal08 @mraarif I have pinned edx-sga in [platform](https://github.com/edx/edx-platform/pull/25316). 
I am trying to find the root cause.

**Comment by awais786** [comment]
It was blocking upgrades PR in platform.

**Pull request merged**